### PR TITLE
Merge Feature/car 67 with Master 

### LINKED ISSUE #65: Detect obstacles on both sides of the car
### Description
As a driver, I want the car to be able to detect obstacles on its left and its right, so that I can have a better idea of the obstacles surrounding the car.

### Acceptance criteria
The car is able to detect obstacles on its left and right.

### LINKED ISSUE #67: Stop when come closer from the side
Description

As a driver, I want the car to stop when an obstacle on its left and its right comes closer, so that I can have a better idea of the obstacles surrounding the car.

Acceptance criteria

The car is able to top when detecting obstacles comes closer on its left and right.

### PR DESCRIPTION
**Pull request by oscarhjern**
This commit will implement driveAroundObsticle-method

### CODE REVIEW

File: arduinoRemote/local.properties
Comment (platisd) [inline arduinoRemote/local.properties:7]: You should remove this file and add it to `.gitignore`

**Review by WhiteMaken — CHANGES_REQUESTED** [review]
Pull request made for Dimitrios. Changes needed before merging, do not merge.

File: car/Car_Control.ino
Comment (platisd) [inline car/Car_Control.ino:107]: I assume there isn't something wrong with this so I am not looking more into it

File: car/car.ino
Comment (platisd) [inline car/car.ino]: Why are you making two separate and different calls to `.getDistance()`?
Same below.

File: car/car.ino
Comment (platisd) [inline car/car.ino]: There is a LOT of duplicated code in these if-else if statements.

File: car/car.ino
Comment (platisd) [inline car/car.ino]: 3 meters is a really really long distance for these sensors. I have found them to be practically unusable when in motion and when tasked to detect obstacles that will not be at a 90 degrees angle compared to the beam they emit.
Therefore, I would strongly advice to use the smallest distance that makes sense for your application, or the default value (which is 100). In other words, unless you need to, don't pass a maximum distance argument yourselves.

**Review by platisd — COMMENTED** [review]
Without knowing **what** exactly doesn't work (you never specified this in the email), these are my first comments.

Additionally, if you want to get a more filtered output from the ultrasonic sensors, you should try the `.getMedianDistance()` method instead of the `getDistance()`.
We can continue the conversation here instead of the emails and other members of the team are definitely welcome to contribute as well.

File: car/car.ino
Comment (oscarhjern) [reply car/car.ino]: Yes. We wanted to make it work, and then make the code neat and nice :)

**Comment by platisd** [comment]
Just got this in an email:

> We have clearly defined steps that the car should do in order to go around the obstacle, and sometimes it does it and sometimes it doesn't. Pls see attached picture for the steps. 
> It seems like it skips steps (rows of code) because it does not execute them. We have tried the lines separately and then it works, so it does not make sense...

Which rows of code are the ones you refer to? You need to help me help you. :smile_cat:

**Comment by oscarhjern** [comment]
> Just got this in an email:
> 
> > We have clearly defined steps that the car should do in order to go around the obstacle, and sometimes it does it and sometimes it doesn't. Pls see attached picture for the steps.
> > It seems like it skips steps (rows of code) because it does not execute them. We have tried the lines separately and then it works, so it does not make sense...
> 
> Which rows of code are the ones you refer to? You need to help me help you. 😸

It is in the "car.ino" file. Row 127-153.

**Comment by platisd** [comment]
> > Just got this in an email:
> > > We have clearly defined steps that the car should do in order to go around the obstacle, and sometimes it does it and sometimes it doesn't. Pls see attached picture for the steps.
> > > It seems like it skips steps (rows of code) because it does not execute them. We have tried the lines separately and then it works, so it does not make sense...
> > 
> > 
> > Which rows of code are the ones you refer to? You need to help me help you. smile_cat
> 
> It is in the "car.ino" file. Row 127-153.

Apply _both_ suggestions (i.e. doing one measurement and using `getMedianDistance`) I've already provided and see if that improves things.

**Comment by majawennroth** [comment]
closing to create a new one with more updates

**Pull request closed without merging**
